### PR TITLE
Auxilliary skin improvements 

### PR DESCRIPTION
**Pull request by daschuer**
Move from Auxiliary,mute to Auxiliary,master, to have an equivalent to talkover.
Adjust Auxiliary,master depending on the configured state. 
fixes https://bugs.launchpad.net/mixxx/+bug/1747212

Streamline the Aux configuration use case by allowing to start the configuration directly from the GUI

### CODE REVIEW

**Comment by daschuer** [comment]
also https://bugs.launchpad.net/mixxx/+bug/1779472

**Comment by ronso0** [comment]
I'll give this a try soonish

**Comment by ronso0** [comment]
LGTM!
just a typo in engineaux.cpp

btw: I find the enable/mute control in Deere hard to discover and not appropriately sized regarding its effect.

**Comment by ronso0** [comment]
Tango (64 Samplers)/skin.xml now has a conflict after merging #1748

File: src/engine/engineaux.cpp [hunk 1]
```diff
@@ -23,9 +23,10 @@ EngineAux::EngineAux(const ChannelHandleAndGroup& handle_group, EffectsManager*
     ControlDoublePrivate::insertAlias(ConfigKey(getGroup(), "enabled"),
                                       ConfigKey(getGroup(), "input_configured"));
 
-    // by default Aux is enabled on the master and disabled on PFL. User
+    // by default Aux is disabled on the master and disabled on PFL. User
     // can over-ride by setting the "pfl" or "master" controls.
-    setMaster(true);
+    // Skins can change that during intalisation, if the master control is not provided.
```
Comment (ronso0) [inline src/engine/engineaux.cpp]: initialisation

**Comment by Be-ing** [comment]
Or maybe we can rename `master` to `master_enabled`?

**Comment by Be-ing** [comment]
I find the current situation as well as the proposed changes in this PR confusing, so I do think there is a good reason to rename Controls.

**Comment by daschuer** [comment]
The CO situation can be confusing if you look at Aux inputs only. These controlls are common for all engine channels in which context they are sensible IMHO.
Sure we can adjust the names. I just had a close look into this, but I am actually fine with the current situation. We may think about a general refactoring along with adjusting and renaming. Everything we do now without it will complicate that. Given the other issues we are working on I want to not the work now. 
This PR is a minimum solution on the current interface. Later changes are always possible. So I think we should merge this without renaming and wait for the right time to issue the remaining topics.

**Comment by daschuer** [comment]
Any further comments?

**Comment by daschuer** [comment]
Conflicts resolved.

**Comment by ronso0** [comment]
Coming back to this, I think we should make the Mic / Aux controls consistent across all skins.
For this it's necessary to
* remove tiny Mute buttons from Deere Mic & Aux units:
no other skin has them, neither does any other deck in Deere
* add a proper Talkover-like Play button to Deere Aux units

@daschuer I can jump in to resolve the conflicts and apply the proposed changes

I'll try to add a `noAuxiliaryInputCnfigured` signal when `[Aux#],master` is toggled On with no Aux device set up, like we have it for the Microphone and Vinyl inputs.

Actually it would be nice to have this Hardware preferences shortcut for unconfigured Headphones, too, so that each device status is visible in skins. But I guess for Headphones it's much more complicated as it affects not just `[..],pfl` but also the Fx routing buttons...

**Comment by ronso0** [comment]
https://github.com/daschuer/mixxx/pull/42

**Comment by ronso0** [comment]
I can send a follow-up PR for https://github.com/mixxxdj/mixxx/pull/43 that adds the Add button for unconfigured mic / aux decks (so we have all related changes here), or I open a separate PR in case yo want to merge this soon.
I don't mind.

**Comment by daschuer** [comment]
@Be-ing: Is this good enough for merge now?

**Comment by ronso0** [comment]
@daschuer Do you mind giving a résumé of the changes in the PR description so everybody knows what this is all about now? [AuxN],master + skin changes + popup for unconfigured Aux

**Comment by daschuer** [comment]
OK, done. 
I think we can merge it to master now, right? 
Does one else want to have a look?

**Comment by ronso0** [comment]
nitpicks: Auxually > Auxilliary in the title ;) so we can find the PR later on. Also no 'mute' controls are exposed in Aux / Mics anymore.

**Comment by daschuer** [comment]
> Also no 'mute' controls are exposed in Aux / Mics anymore.

This was my intention. Do you think it is wrong? I don't think we need both controls in the skin.

**Comment by ronso0** [comment]
> > Also no 'mute' controls are exposed in Aux / Mics anymore.
> 
> This was my intention. Do you think it is wrong? I don't think we need both controls in the skin.

No, of curse not, it's alright. I just stumbled over the description, that's all:
Adjust the skin start-up to Aux silence, depending of mute or master is exposed. But we don't set anything Aux/Micrelated in skins anymore, because now c++ code takes care of muting external inputs.

**Comment by daschuer** [comment]
Ah OK. Hopefully fixed now.

**Comment by daschuer** [comment]
@ronso0 Can we merge this, or want someone else have a look?

**Comment by Be-ing** [comment]
It looks like my [concern above](https://github.com/mixxxdj/mixxx/pull/1755#issuecomment-450035196) was never addressed? I think it is important to resolve that before merging this. If we merge this as is, people will map controllers to the awkwardly named ControlObject and we will be committed to it for backwards compatibility.

**Comment by ronso0** [comment]
Previous sitution:
`[MicN], talkover` (muted) + `[AuxN],mute` (0 on startup)
`[AuxN],mute` could be toggled regardless of a respective sound device being set up or not.

Improvement of this PR:
`[MicN], talkover` + `[AuxN],master` (booth muted on startup)
Both controls open a question dialog now asking for configuration of an input device.
No need to set any Mic /Aux controls in skins on startup.

The CO names in master for configured sound devices are incosistent anyway:
[Master],booth_enabled
[Master],headEnabled
[MicN] / [AuxN],enabled
[Vinylcontrol],input_configured

This PR aims to streamline the Aux workflow and use the same controls for that in all skins.
Also it unclutters skins by showing a 'Add' button instead of entire unconfigured Mic / Aux decks.
How this is now implemented is way beyond the initial idea anyway, so I think adjusting all input and output related control names here is out of scope.

Btw, I'd prefer `[In/OutputX],configured` instead of `enabled`.

**Comment by Holzhaus** [comment]
I'm all for improving the CO names, but IMHO this is beyond the scope of this PR. I'd say let's merge this.

**Comment by ronso0** [comment]
I'll take care of the conflicts.

**Comment by ronso0** [comment]
https://github.com/daschuer/mixxx/pull/48

**Comment by ronso0** [comment]
@daschuer Please take a look at my PR resolving the conflicts.
With that this here is again ready to be merged.

**Comment by daschuer** [comment]
Done. Thank you for the reminder.

**Comment by daschuer** [comment]
@Be-ing: Can you have another look here?

**Comment by ronso0** [comment]
@daschuer 39e983e0e92d409919d22934289c9b36a9f779e7 adds the new TooltipId from #2379 
to 'Configure' buttons.

**Comment by ronso0** [comment]
bce01af58889764f2adcc765758aabc8d1e62845 merges master & fixes the conflict in Deere/auxiliary.xml

**Comment by ronso0** [comment]
@Be-ing Do you find time to review this?
@daschuer with the patch I posted above it's ready for master.

**Comment by daschuer** [comment]
Your commits are added. Sorry I forget about it.  

>> EDIT Oha, seems I have to rebase on master..

No. the merge it is still working cleanly.

**Comment by ronso0** [comment]
I fixed some TooltipIds, and I'd merge this if there are now major objections. @Be-ing ?

**Comment by Holzhaus** [comment]
There's trailing whitespace and a few other issues. Can you fix them via pre-commit or using the diff from the Travis pre-commit-pr job?

**Comment by ronso0** [comment]
me or @daschuer ?
I'm travelling for a job right now so I won't find the time to work out how to push to this PR or grab the diff from pre-commit and apply, sry.

**Comment by Holzhaus** [comment]
> me or @daschuer ?
> I'm travelling for a job right now so I won't find the time to work out how to push to this PR or grab the diff from pre-commit and apply, sry.

It was directed at @daschuer then :smiley:

**Comment by daschuer** [comment]
Done!

**Comment by ronso0** [comment]
Great, thx!

**Pull request merged**